apollo-router/src/spec/query.rs

```diff
@@ -146,18 +148,6 @@ impl Query {
                                 errors: Vec::new(),
                                 nullified: Vec::new(),
                             };
-                            // Detect if root __typename is asked in the original query (the qp doesn't put root __typename in subselections)
-                            // cf https://github.com/apollographql/router/issues/1677
-                            let operation_kind_if_root_typename =
-                                original_operation.and_then(|op| {
-                                    op.selection_set
-                                        .iter()
-                                        .any(|f| f.is_typename_field())
-                                        .then(|| *op.kind())
-                                });
-                            if let Some(operation_kind) = operation_kind_if_root_typename {
-                                output.insert(TYPENAME, operation_kind.default_type_name().into());
-                            }
```

Comment:
We don't need it anymore since I changed code here:
https://github.com/apollographql/router/pull/6009/files?diff=split&w=1#diff-ca21c6c2d8f5eacbc8fd08524c17571552b1ec649c2911b908b03b94b2c7c99fR793

Reply:
good, that part was really fragile